zproject/backends.py

```diff
@@ -489,6 +500,85 @@ def authenticate(self, dev_auth_username: Optional[str]=None, realm: Optional[Re
             return None
         return common_get_active_user(dev_auth_username, realm, return_data=return_data)
 
+class MediaWikiAuthBackend(SocialAuthMixin, MediaWiki):
+    auth_backend_name = "MediaWiki"
+
+    def get_email_address(self, *args, **kwargs):
+        # type: (*Any, **Any) -> Optional[Text]
+        try:
+            return kwargs['response']['email']
+        except KeyError:
+            return None
+
+    def get_full_name(self, *args, **kwargs):
+        # type: (*Any, **Any) -> Text
+        try:
+            name = kwargs['response']['realname'] or kwargs['response']['username']
+        except KeyError:
+            name = ''
+
+        return name
+
+    def get_authenticated_user(self, *args: Any, **kwargs: Any) -> Optional[UserProfile]:
+        """Finish the auth process once the access_token was retrieved"""
+        access_token = cast(Dict[Text, Text], args[0])
+        data = self.get_user_details(access_token)
+        if data is not None and 'access_token' not in data:
+            data['access_token'] = access_token
+        kwargs.update({'response': data, 'backend': self})
+        return self.strategy.authenticate(*args, **kwargs)
+
+    def get_user_details(self, access_token: Dict[Text, Text]) -> Dict[Text, Any]:
+        """
+        Gets the user details from Special:OAuth/identify
+        """
+        key, secret = self.get_key_and_secret()
```

Comment:
This copies an awful lot of code from `MediaWiki` backend.  It is BSD code, so it is probably legal to copy, but it heavily increases the maintenance overhead.

I think you can call super() with a fake `response['access_token']` so it works correctly.

Reply:
That is correct, however, the reason it had to be done was that we needed to return `identity`; upstream returns
```
return {
            'username': identity['username'],
            'userID': identity['sub']
}
```
which doesn't serve our purpose. I'll add a comment to explain this.

Reply:
Ok. I have added a comment to explain the need to override this function.